src/new-components/modal/modal.test.tsx

```diff
@@ -1,10 +1,42 @@
 import * as React from 'react'
-import { render, screen, within } from '@testing-library/react'
+import { act, render, screen, within } from '@testing-library/react'
 import { Modal, ModalHeader, ModalFooter, ModalActions, ModalBody, ModalCloseButton } from './modal'
 import userEvent from '@testing-library/user-event'
 import { axe } from 'jest-axe'
 
+// Ariakit's dialog performs async updates and must be wrapped in an act to prevent warnings
+function renderModal(children: React.ReactElement) {
```

Comment:
@Bloomca to prevent act warnings yes. It is an existing problem though, see Pedro's notes in https://github.com/Doist/todoist-web/pull/5614. Since we've got the jest console reporter we can track our efforts in bringing them down.

Reply:
I see, thanks for the info!

Reply:
Wow, this sucks 😕

I wonder if we now should provide this `renderModal` helper (with a more generic name) to be THE way to render anything in tests. It does not seem to have anything tied to the fact that is rendering a modal, right?

Though, I'm not 100% comfortable assuming that doing this in every test call to `render` is not going to have any unwanted consequences.